### DISCUSSION TITLE
Use Oracle sys_guid in pydantic

### DISCUSSION THREAD

**Discussion by pylkey** [pydantic/pydantic#3958] [Question]
Hello everyone.
From db to Pydantic i get binary data. This data is Oracle sys_guid id. I need change binary data to str by Pydantic.
I tried ```pydantic.UUID4``` , and i get error ```uuid version 4 expected (type=value_error.uuid.version; required_version=4)```.
For example:
```
class News(BaseModel):
    id:UUID4
    title:str
```
I tried ```@validator```, and i get error``` 'utf-8' codec can't decode byte 0xb6 in position 2: invalid start byte (type=value_error.unicodedecode)``` too.
For example:
```
class Report(BaseModel):
    id:str
    title:str

    @validator('id')
    def hexlify_id(cls, value):
        return value.hex()
```
How use sys_guid in sqlalchemy write [here](https://www.popmartian.com/tipsntricks/2017/04/12/using-oracle-sys_guid-and-raw-in-sqlalchemy/) and this work fine. But i need help with Pydantic, how convert Oracle sys_guid to str and str to Oracle sys_guid

**Comment by samuelcolvin** [comment]
You need to provide a smaller self contained example.

There's too much code here.

**Reply by pylkey** [reply]
Got it.
